### ISSUE TITLE
What do I do when I finish reading the book?

### ISSUE THREAD

**Issue by artillect** [intermezzOS/book#154]
I've completed everything covered in the book, what can I do next? It doesn't seem like anything has been written for the book in a while. What resources can I read to continue working on the OS?

(I know that Github isn't exactly the right place, but I've looked around and asked the IRC and have found nothing. It seems like the Github is slightly more active.)

**Comment by bjorn3** [comment]
Have you read http://os.phil-opp.com/ ?

**Comment by steveklabnik** [comment]
Yeah I haven't had a ton of time for the book lately. The kernel is farther along, though. So one thing is "go check out the kernel code and see if you can understand it."

I second @bjorn3 's recommendation of Phil's blog, it's great.

**Comment by artillect** [comment]
@bjorn3 @steveklabnik Thank you for the recommendations, I'll definitely check those out. It seems like Phil's blog has also not been updated in a few months, but it definitely has a lot more content to work off on.

**Comment by steveklabnik** [comment]
Other random resources I like

https://pdos.csail.mit.edu/6.828/2014/xv6.html

http://littleosbook.github.io/

http://pages.cs.wisc.edu/~remzi/OSTEP/

they're not rust, but they're good!

**Comment by phil-opp** [comment]
@Artillect:

> It seems like Phil's blog has also not been updated in a few months

There have been [some smaller updates](https://github.com/phil-opp/blog_os/pulls?q=is%3Apr+is%3Aclosed+label%3Arelnotes) over the last weeks, so everything should be up-to-date. And the next post is almost ready (maybe even today :) ).

@steveklabnik xv6 looks really interesting, thanks for the link!
